### DISCUSSION TITLE
A way to share bookmarks

### DISCUSSION THREAD

**Discussion by mbalaji-atlas** [alefragnani/vscode-bookmarks#603] [Ideas]
I find labeled bookmarks to be a nice way to guide a user through the codebase. Why not docs? Because this way user does not leave from codebase while following steps to do "something" (less context switching). So problem I face is how to transfer bookmarks from my vscode to my teammates vscode ? (in same version controlled project ofc)

Though the main problem to solve is this (considering this is a problem worth solving), there would be bunch of solutions that can solve this problem. But the most simplest one to start with (I assume) would be a way to import/export bookmarks from a project.

APIs:
- Export bookmarks to a json file
- Import bookmarks from json file
   - throw error if cannot find the particular files or lines of bookmark in the current project

**Comment by alefragnani** [comment]
Hi @mbalaji-atlas ,

If you want to _share_ bookmarks between your teammates, you could simply use `bookmarks.saveBookmarksInProject` setting as `true`, and the extension will store the bookmarks within the project, in `.vscode/bookmarks.json` file. Doing so, every time someone pulls the repo, the bookmarks will be there.

Right now, there is no export/import feature, only an open issue asking for it.

BTW, if you want _guidance_ on your codebase, _following steps to do something_, maybe you should take a look at the [CodeTour extension](https://marketplace.visualstudio.com/items?itemName=vsls-contrib.codetour#recording-tours). It is a great extension too.

Hope this helps
